parsil/src/utils.rs

```diff
@@ -115,8 +115,10 @@ pub struct PlaceholderSettings {
     pub min_block_placeholder: String,
     /// The placeholder for the maximal value of the primary index
     pub max_block_placeholder: String,
-    /// The number of free-standing `$i` placeholders
+    /// The maximum number of free-standing `$i` placeholders
     pub max_free_placeholders: usize,
+    /// The number of provided parameters for placeholders, if any
+    pub provided_parameters: Option<usize>,
```

Comment:
Should that rather be `placeholders_count`? And be a [`OnceCell`](https://doc.rust-lang.org/beta/std/cell/struct.OnceCell.html) instead?